server/controllers/nlp.js

```diff
@@ -77,6 +102,27 @@ async function executeFunctions(history, dataset, errorCallback) {
     return currentData
 }
 
+/**
+ * takes a tokenHolder and extracts (recursively) all information that is needed after classification
+ */
+function simplifyTokenHolder(tokenHolder) {
+    let data = {
+        "token": tokenHolder.token,
+        "label": tokenHolder.label,
+        "labelType": tokenHolder.labelType
+    };
+    if (tokenHolder.hasOwnProperty("column") && tokenHolder.column !== undefined) {
+        data.column = tokenHolder.column
```

Comment:
Shouldnt we delete the column from the column array, if we find it with a filter? Otherwise the "show a histogram of all departments with age greater than 50" detects a two column histogram which is not correct?

Reply:
i have had this
`show a histogram of age where it is bigger than 20`
where i remove age here (because we find it with the filter) then it this would be wrong as well...
'I have not had a good idea here...